src/solver/batchnorm/forward_training_ck.cpp

```diff
@@ -174,68 +158,397 @@ static ConvSolution MakeAnyInvokerFactory(const miopen::batchnorm::ProblemDescri
+{
+    const auto& args       = CKArgsBNormFwdTraining{problem};
+    const auto bn_fwd_ptrs = DeviceOpBNFwdTrainingPtrs<XDataType,
+                                                       YDataType,
+                                                       AccDataType,
+                                                       ScaleDataType,
+                                                       BiasDataType,
+                                                       MeanVarDataType>::GetInstances();
+    assert(!bn_fwd_ptrs.empty());
+    int count = 0;
+    for(const auto& it : bn_fwd_ptrs)
+    {
+        auto argument_ptr = it->MakeArgumentPointer(args.xyLengths,
+                                                    args.xyStrides,
+                                                    args.xyStrides,
+                                                    args.reduceDims,
+                                                    args.arrScaleBiasMeanVarLengths,
+                                                    args.arrScaleBiasMeanVarStrides,
+                                                    args.arrScaleBiasMeanVarStrides,
+                                                    args.arrScaleBiasMeanVarStrides,
+                                                    nullptr,
+                                                    nullptr,
+                                                    nullptr,
+                                                    0.0,
+                                                    PassThroughOp{},
+                                                    nullptr,
+                                                    nullptr,
+                                                    nullptr,
+                                                    0.0,
+                                                    nullptr,
+                                                    nullptr);
+        if(it->IsSupportedArgument(argument_ptr.get()))
+        {
+            return count;
+        }
+        count++;
+    }
+    return -1;
+}
+
+template <typename XDataType,
+          typename YDataType,
+          typename AccDataType,
+          typename ScaleDataType,
+          typename BiasDataType,
+          typename MeanVarDataType>
+ConvSolution InvokerFactoryMakerNHWC(const miopen::batchnorm::ProblemDescription& bn_problem,
+                                     const PerformanceConfigBnCKFwdTraining& config)
+{
+    ConvSolution result;
+    auto bn_fwd_ptrs = DeviceOpBNFwdTrainingPtrs<XDataType,
+                                                 YDataType,
+                                                 AccDataType,
+                                                 ScaleDataType,
+                                                 BiasDataType,
+                                                 MeanVarDataType>::GetInstances();
+
+    assert(config.index >= 0 && !bn_fwd_ptrs.empty() && config.index < bn_fwd_ptrs.size());
```

Comment:
This should probably throw. But if the asserts above are changed to throws, assert here is probably manageable.